### PR TITLE
fix: handle multibyte chars properly in gzip decoder

### LINKED ISSUE #132: StringClient gunzip not safe for multibyte characters
StringClient directly decodes chunks to utf8 strings which is not safe in the case of a chunk boundary falling within a multibyte character. Instead it should be using the string_decoder api which accounts for this. https://nodejs.org/api/string_decoder.html

https://github.com/restify/clients/blob/7f98f44c484e5fb4cb64cffa3e04ea8d4bb8d309/lib/StringClient.js#L220

### PR DESCRIPTION
**Pull request by DonutEspresso**
Fixes #132

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,7 +49,7 @@
     "coveralls": "^2.11.4",
     "eslint": "^4.1.1",
     "istanbul": "^0.4.0",
-    "jscs": "^3.0.0",
+    "jscs": "^2.11.0",
```
Comment (DonutEspresso) [inline package.json]: 3.x seems to hang freeze for some reason. Should think about moving to eslint for style linting too at some point.

Comment (retrohacker) [reply package.json]: :+1: for moving to eslint, jscs is no longer supported. But that is outside the scope of this PR :-)

File: test/etc/multibyte.txt [hunk 2]
```diff
@@ -0,0 +1 @@
...的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件遙遠未來的事件
```
Comment (retrohacker) [inline test/etc/multibyte.txt]: heh, nice test!

Comment (retrohacker) [reply test/etc/multibyte.txt]: Oh my, didn't think that commenting on this would expand the whole line.

**Comment by DonutEspresso** [comment]
Rebased against master - will wait for add'l feedback from peanut gallery.

**Pull request merged**